src/panel_state_manager.cpp

```diff
@@ -1095,6 +1095,52 @@ void PanelStateManager::setCEState()
         }
     }
 }
+
+bool PanelStateManager::isFunctionEnabled(const types::FunctionNumber funcNum)
+{
+    auto pos = find_if(panelFunctions.begin(), panelFunctions.end(),
+                       [funcNum](const PanelFunctionality& afunctionality) {
+                           return afunctionality.functionNumber == funcNum;
+                       });
+    if (pos != panelFunctions.end())
+    {
+        return pos->functionActiveState;
+    }
+
+    return false;
+}
+
+types::ReturnStatus PanelStateManager::triggerFunctionDirectly(
+    const types::FunctionNumber funcNum)
+{
+    const auto funcStatus = isFunctionEnabled(funcNum);
+
+    if (funcStatus)
+    {
+        return (funcExecutor->executeFunctionDirectly(funcNum));
+    }
+
+    std::cerr << "Function " << static_cast<int>(funcNum) << " is disabled."
```

Comment:
funcStatus variable not required.
Also, the log should move in the function "isFunctionEnabled".
So that all caller need not have this log (in case in future there are more callers to api "isFunctionEnabled")

Reply:
done